packages/client/hmi-client/src/components/model/stockflow/tera-stockflow-tables.vue

```diff
@@ -0,0 +1,265 @@
+import Button from 'primevue/button';
+
+const props = defineProps<{
+	model: Model;
+	modelConfigurations?: ModelConfiguration[];
+	readonly?: boolean;
+}>();
+
+const emit = defineEmits(['update-model']);
+
+const mmt = ref<MiraModel>(emptyMiraModel());
+const mmtParams = ref<MiraTemplateParams>({});
+
+const transientModel = ref(cloneDeep(props.model));
+const initialsLength = computed(() => props.model?.semantics?.ode?.initials?.length ?? 0);
+const parametersLength = computed(
+	() =>
+		(props.model?.semantics?.ode.parameters?.length ?? 0) +
+		(props.model?.model?.auxiliaries?.length ?? 0)
+);
+const observables = computed(() => props.model?.semantics?.ode?.observables ?? []);
+const time = computed(() =>
+	props.model?.semantics?.ode?.time ? [props.model?.semantics.ode.time] : []
+);
+const extractions = computed(() => {
+	const attributes = props.model?.metadata?.attributes ?? [];
+	return groupBy(attributes, 'amr_element_id');
+});
+const stocks = computed(() => props.model?.model?.stocks ?? []);
+const flows = computed(() => props.model?.model?.flows ?? []);
+const otherConcepts = computed(() => {
+	const ids = [...(stocks.value?.map((s) => s.id) ?? []), ...(flows.value?.map((f) => f.id) ?? [])];
+
+	// find keys that are not aligned
+	const unalignedKeys = Object.keys(extractions.value).filter((k) => !ids.includes(k));
+
+	let unalignedExtractions: Dictionary<any>[] = [];
+	unalignedKeys.forEach((key) => {
+		unalignedExtractions = unalignedExtractions.concat(
+			extractions.value[key.toString()].filter((e) =>
+				['anchored_extraction', 'anchored_entity'].includes(e.type)
+			)
+		);
+	});
+
+	return unalignedExtractions ?? [];
+});
+
+const updateInitial = (inits: Initial[]) => {
+	const modelInitials = transientModel.value.semantics?.ode.initials ?? [];
+	for (let i = 0; i < modelInitials.length; i++) {
+		const foundInitial = inits.find((init) => init.target === modelInitials![i].target);
+		if (foundInitial) {
+			modelInitials[i] = foundInitial;
+		}
+	}
+};
+
+const updateParam = (params: ModelParameter[]) => {
+	const modelParameters = transientModel.value.semantics?.ode?.parameters ?? [];
+	for (let i = 0; i < modelParameters.length; i++) {
+		const foundParam = params.find((p) => p.id === modelParameters![i].id);
+		if (foundParam) {
+			modelParameters[i] = foundParam;
+		}
+	}
+	// FIXME: Sometimes auxiliaries can share the same ids as parameters so for now both are be updated in that case
+	const modelAuxiliaries = transientModel.value.model?.auxiliaries ?? [];
+	for (let i = 0; i < modelAuxiliaries.length; i++) {
+		const foundParam = params.find((p) => p.id === modelAuxiliaries![i].id);
+		if (foundParam) {
+			modelAuxiliaries[i] = foundParam;
+		}
+	}
+};
+
+watch(
+	() => props.model,
+	async (model) => {
+		if (!model) return;
+		transientModel.value = cloneDeep(model);
+		const response: any = await getMMT(model);
+		mmt.value = response.mmt;
+		mmtParams.value = response.template_params;
```

Comment:
If the transient model changes, how do we keep the mmt and mmtParams in sync?

Reply:
Do we want to keep the MMT in sync with the `transientModel` before the edits are saved? I think it makes sense to just sync with the `model` once it's properly updated.